collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -26,37 +29,66 @@
 import org.odk.collect.android.tasks.SaveToDiskResult;
 import org.odk.collect.android.utilities.FileUtils;
 import org.odk.collect.android.utilities.MediaUtils;
+import org.odk.collect.android.utilities.QuestionMediaManager;
 import org.odk.collect.utilities.Clock;
 
 import java.io.File;
 
 import timber.log.Timber;
+
+import java.util.Collection;
 import java.util.HashMap;
+import java.util.Map;
 
 import static org.odk.collect.android.tasks.SaveFormToDisk.SAVED;
 import static org.odk.collect.android.tasks.SaveFormToDisk.SAVED_AND_EXIT;
 import static org.odk.collect.android.utilities.StringUtils.isBlank;
 
-public class FormSaveViewModel extends ViewModel implements ProgressDialogFragment.Cancellable, RequiresFormController {
+public class FormSaveViewModel extends ViewModel implements ProgressDialogFragment.Cancellable, RequiresFormController, QuestionMediaManager {
+    public static final String ORIGINAL_FILES = "originalFiles";
+    public static final String RECENT_FILES = "recentFiles";
 
+    private final SavedStateHandle stateHandle;
     private final Clock clock;
     private final FormSaver formSaver;
+    private final MediaUtils mediaUtils;
 
-    private String reason = "";
     private final MutableLiveData<SaveResult> saveResult = new MutableLiveData<>(null);
+    private String reason = "";
+
+    private Map<String, String> originalFiles = new HashMap<>();
+    private Map<String, String> recentFiles = new HashMap<>();
 
     @Nullable
     private FormController formController;
 
     @Nullable
-    private AsyncTask saveTask;
+    private AsyncTask<Void, String, SaveToDiskResult> saveTask;
 
     private final Analytics analytics;
 
-    public FormSaveViewModel(Clock clock, FormSaver formSaver, Analytics analytics) {
+    public FormSaveViewModel(SavedStateHandle stateHandle, Clock clock, FormSaver formSaver, MediaUtils mediaUtils, Analytics analytics) {
+        this.stateHandle = stateHandle;
         this.clock = clock;
         this.formSaver = formSaver;
+        this.mediaUtils = mediaUtils;
         this.analytics = analytics;
+
+        if (stateHandle.get(ORIGINAL_FILES) != null) {
+            originalFiles = stateHandle.get(ORIGINAL_FILES);
+        }
+        if (stateHandle.get(RECENT_FILES) != null) {
+            recentFiles = stateHandle.get(RECENT_FILES);
+        }
+    }
+
+    @Override
+    protected void onCleared() {
+        originalFiles.clear();
```

Comment:
I don't think we need to clear these variables out here. They are part of the ViewModel which will be removed from memory.